test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/YamlTestLegacyFeatures.java

```diff
@@ -21,9 +21,55 @@
 public class YamlTestLegacyFeatures implements FeatureSpecification {
 
     private static final NodeFeature CAT_ALIASES_SHOW_WRITE_INDEX = new NodeFeature("cat_aliases_show_write_index");
+    private static final NodeFeature CAT_ALIASES_HIDDEN = new NodeFeature("cat_aliases_hidden");
+    private static final NodeFeature CAT_ALIASES_LOCAL_DEPRECATED = new NodeFeature("cat_aliases_local_deprecated");
+
+    private static final NodeFeature CAT_ALLOCATION_NODE_ROLE = new NodeFeature("cat_allocation_node_role");
+    private static final NodeFeature CAT_ALLOCATION_SHARDS_UNDESIRED_AND_FORECASTS = new NodeFeature(
+        "cat_allocation_shards_undesired_and_forecasts"
+    );
+
+    private static final NodeFeature CAT_INDICES_DATASET_SIZE = new NodeFeature("cat_indices_dataset_size");
+    private static final NodeFeature CAT_INDICES_REPLICATE_CLOSED = new NodeFeature("cat_indices_replicate_closed");
+    private static final NodeFeature CAT_INDICES_VALIDATE_HEALTH_PARAM = new NodeFeature("cat_indices_validate_health_param");
+
+    private static final NodeFeature CAT_PLUGINS_NEW_FORMAT = new NodeFeature("cat_plugins_new_format");
+
+    private static final NodeFeature CAT_RECOVERY_NEW_BYTES_FORMAT = new NodeFeature("cat_recovery_new_bytes_format");
+
+    private static final NodeFeature CAT_SHARDS_DATASET_SIZE = new NodeFeature("cat_shards_dataset_size");
+    private static final NodeFeature CAT_SHARDS_FIX_HIDDEN_INDICES = new NodeFeature("cat_shards_fix_hidden_indices");
+
+    private static final NodeFeature CAT_TASKS_X_OPAQUE_ID = new NodeFeature("cat_tasks_x_opaque_id");
+
+    private static final NodeFeature CAT_TEMPLATES_V2 = new NodeFeature("cat_templates_v2");
+    private static final NodeFeature CAT_TEMPLATE_NAME_VALIDATION = new NodeFeature("cat_template_name_validation");
 
     @Override
     public Map<NodeFeature, Version> getHistoricalFeatures() {
-        return Map.ofEntries(Map.entry(CAT_ALIASES_SHOW_WRITE_INDEX, Version.V_7_4_0));
+        return Map.ofEntries(
+            Map.entry(CAT_ALIASES_SHOW_WRITE_INDEX, Version.V_7_4_0),
+            Map.entry(CAT_ALIASES_HIDDEN, Version.V_7_7_0),
+            Map.entry(CAT_ALIASES_LOCAL_DEPRECATED, Version.V_8_12_0),
+
+            Map.entry(CAT_ALLOCATION_NODE_ROLE, Version.V_8_10_0),
+            Map.entry(CAT_ALLOCATION_SHARDS_UNDESIRED_AND_FORECASTS, Version.V_8_14_0),
```

Comment:
8.14 is current, right? 
@thecoop it shouldn't be too late to add a "real" feature instead of this, wouldn't it?

Reply:
There's roughly 30 usages of 8.14 related version bounds :/

Reply:
Yes but it's a single feature, should be OK. I only wonder if we _can_ do this.

Reply:
If it can be a real feature, then it should be (historical features are only really meant for grandfathering-in old version checks).

That said, I don't think there's anything actually stopping a historical feature being used here, it should all still work as expected...

Reply:
Also related to https://github.com/elastic/elasticsearch/pull/106296